json_object.c

```diff
@@ -236,12 +241,15 @@ enum json_type json_object_get_type(const struct json_object *jso)
 }
 
 void* json_object_get_userdata(json_object *jso) {
-	return jso->_userdata;
+	return jso == NULL ? NULL : jso->_userdata;
```

Comment:
Largely a matter of style, but also compilers often optimize on the basis that the "true" part of a conditional is the most likely.

This would probably be more efficient if reversed, i.e.

    return jso ? jso->_userdata : NULL;

or even 

    return jso ? jso->_userdata : jso;